django_auth0/models.py

```diff
@@ -1 +1,388 @@
+
+    def get_or_create(self, **kwargs):
+        try:
+            return self.get(**kwargs), False
+        except self.model.DoesNotExist:
+            return self.create(**kwargs), True
+
+    def update_or_create(self, defaults=None, **kwargs):
+        defaults = defaults or {}
+        try:
+            user = self.get(**kwargs)
+        except self.model.DoesNotExist:
+            kwargs.update(defaults)
+            return self.create(**kwargs), True
+
+        if 'password' in defaults:
+            user.set_password(defaults.pop('password'))
+        for k, v in defaults.items():
+            setattr(user, k, v)
+        user.save(using=self._db)
+        user._remove_raw_password()
+        return user, False
+
+    def _create_user(self, email, given_name, surname, password):
+        if not email:
+            raise ValueError("Users must have an email address")
+
+        if not given_name or not surname:
+            raise ValueError("Users must provide a given name and a surname")
+
+        user = self.model(email=Auth0UserManager.normalize_email(email),
+            given_name=given_name, surname=surname)
+
+        user.set_password(password)
+        user.save(using=self._db)
+        user._remove_raw_password()
+        return user
+
+    def create_user(self, email, given_name=None, surname=None, password=None,
+                    first_name=None, last_name=None):
+        if first_name and not given_name:
+            given_name = first_name
+        if last_name and not surname:
+            surname = last_name
+
+        return self._create_user(email=email, given_name=given_name, surname=surname,
+                          password=password)
+
+    def create_superuser(self, **kwargs):
+        user = self.create_user(**kwargs)
+        user.is_admin = True
+        user.is_staff = True
+        user.is_superuser = True
+        user.save(using=self._db)
+        user._remove_raw_password()
+        return user
+
+    def delete(self, *args, **kwargs):
+        for user in self.get_queryset():
+            user.delete(*args, **kwargs)
+
+        # Clear the result cache, in case this QuerySet gets reused.
+        self._result_cache = None
+
+    def sync_accounts_from_stormpath(self, sync_groups=True):
+        """ :arg sync_groups: WARNING!!! Groups will be deleted from stormpath
+                                if not present locally when user logs in!
+        Sync accounts from stormpath -> local database.
+        This may take a long time, depending on how many users you have in your
+        Stormpath application. It also makes numerous database queries.
+        This method updates local users from stormpath or creates new ones
+        where the user does not exist locally. This is an additive operation,
+        meaning it should delete no data from the local database OR stormpath.
+        """
+        if sync_groups:
+            sp_groups = [g.name for g in APPLICATION.groups]
```

Comment:
I see you took it from stormpath 
```
APPLICATION = CLIENT.applications.get(settings.STORMPATH_APPLICATION)
```
It is going to brake things, may consider refactoring this.